s3-source-connector/src/test/java/io/aiven/kafka/connect/s3/source/utils/FileReaderTest.java

```diff
@@ -32,82 +28,196 @@
 import com.amazonaws.services.s3.model.ListObjectsV2Request;
 import com.amazonaws.services.s3.model.ListObjectsV2Result;
 import com.amazonaws.services.s3.model.S3ObjectSummary;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.mockito.Mock;
 
 class FileReaderTest {
 
     private static final String TEST_BUCKET = "test-bucket";
-    @Mock
     private AmazonS3 s3Client;
 
-    @Mock
     private OffsetManager offsetManager;
 
     private FileReader fileReader;
 
-    private Map<String, String> properties;
-
-    @BeforeEach
-    public void setUp() {
-        properties = new HashMap<>();
-        setBasicProperties();
-        final S3SourceConfig s3SourceConfig = new S3SourceConfig(properties);
-        offsetManager = mock(OffsetManager.class);
-        fileReader = new FileReader(s3SourceConfig, TEST_BUCKET, Collections.emptySet());
-        s3Client = mock(AmazonS3.class);
+    private static Map<String, Object> getConfigMap(final int maxTasks, final int taskId) {
+        final Map<String, Object> configMap = new HashMap<>();
+        configMap.put("tasks.max", String.valueOf(maxTasks));
+        configMap.put("task.id", String.valueOf(taskId));
+        return configMap;
     }
 
     @Test
-    void testFetchObjectSummariesWithNoObjects() throws IOException {
+    void testFetchObjectSummariesWithNoObjects() {
+        initializeWithTaskConfigs(3, 1);
         final ListObjectsV2Result listObjectsV2Result = createListObjectsV2Result(Collections.emptyList(), null);
         when(s3Client.listObjectsV2(any(ListObjectsV2Request.class))).thenReturn(listObjectsV2Result);
         when(offsetManager.getOffsets()).thenReturn(new HashMap<>());
 
         final Iterator<S3ObjectSummary> summaries = fileReader.fetchObjectSummaries(s3Client);
-        assertFalse(summaries.hasNext());
+        assertThat(summaries.hasNext()).isFalse();
     }
 
     @Test
-    void testFetchObjectSummariesWithOneNonZeroByteObject() throws IOException {
-        final S3ObjectSummary objectSummary = createObjectSummary(1, "key1");
-        final ListObjectsV2Result listObjectsV2Result = createListObjectsV2Result(
-                Collections.singletonList(objectSummary), null);
-        when(s3Client.listObjectsV2(any(ListObjectsV2Request.class))).thenReturn(listObjectsV2Result);
-        when(offsetManager.getOffsets()).thenReturn(new HashMap<>());
+    void testFetchObjectSummariesWithOneObjectWithBasicConfig() {
+        final String objectKey = "any-key";
 
-        final Iterator<S3ObjectSummary> summaries = fileReader.fetchObjectSummaries(s3Client);
+        initializeWithTaskConfigs(1, 0);
+        final Iterator<S3ObjectSummary> summaries = getS3ObjectSummaryIterator(objectKey);
+        assertThat(summaries.hasNext()).isTrue();
+        assertThat(summaries.next().getSize()).isEqualTo(1);
+    }
+
+    @Test
+    void testFetchObjectSummariesWithOneNonZeroByteObjectWithTaskId1() {
```

Comment:
all this tests from here up to line 188 could be a single parameterized test

Reply:
done.